std/algebra/native/sw_bls12377/hints.go

```diff
@@ -18,44 +20,71 @@ func init() {
 	solver.RegisterHint(GetHints()...)
 }
 
-func decomposeScalarG1(scalarField *big.Int, inputs []*big.Int, res []*big.Int) error {
+func decomposeScalarG1Simple(scalarField *big.Int, inputs []*big.Int, outputs []*big.Int) error {
+	if len(inputs) != 1 {
+		return fmt.Errorf("expecting one input")
+	}
+	if len(outputs) != 2 {
+		return fmt.Errorf("expecting two outputs")
+	}
+	cc := getInnerCurveConfig(scalarField)
+	sp := ecc.SplitScalar(inputs[0], cc.glvBasis)
+	outputs[0].Set(&(sp[0]))
+	outputs[1].Set(&(sp[1]))
+
+	return nil
+}
+
+func decomposeScalarG1(scalarField *big.Int, inputs []*big.Int, outputs []*big.Int) error {
```

Comment:
For `jointScalarMulUnsafe` we still need the old version that sets either of the top bits of the GLV sub-scalars. The loopkup-based new version is not efficient for `jointScalarMulUnsafe` as it would require multiplexers which are costly compared to native point arithmetic.

Reply:
I see!